nnvm/python/nnvm/testing/yolo_detection.py

```diff
@@ -65,72 +65,104 @@ def _box_union(a, b):
 def _box_iou(a, b):
     return _box_intersection(a, b)/_box_union(a, b)
 
-def get_region_boxes(layer_in, imw, imh, netw, neth, thresh, probs,
-                     boxes, relative, tvm_out):
-    "To get the boxes for the image based on the prediction"
-    lw = layer_in.w
-    lh = layer_in.h
-    probs = [[0 for i in range(layer_in.classes + 1)] for y in range(lw*lh*layer_in.n)]
-    boxes = [Box(0, 0, 0, 0) for i in range(lw*lh*layer_in.n)]
-    for i in range(lw*lh):
-        row = int(i / lw)
-        col = int(i % lw)
-        for n in range(layer_in.n):
-            index = n*lw*lh + i
-            obj_index = _entry_index(0, lw, lh, layer_in.outputs, layer_in.classes,
-                                     layer_in.coords, n*lw*lh + i, layer_in.coords)
-            box_index = _entry_index(0, lw, lh, layer_in.outputs, layer_in.classes,
-                                     layer_in.coords, n*lw*lh + i, 0)
-            mask_index = _entry_index(0, lw, lh, layer_in.outputs, layer_in.classes,
-                                      layer_in.coords, n*lw*lh + i, 4)
-            scale = 1 if layer_in.background  else tvm_out[obj_index]
-            boxes[index] = _get_region_box(tvm_out, layer_in.biases, n, box_index, col,
-                                           row, lw, lh, lw*lh)
-            if not layer_in.softmax_tree:
-                max_element = 0
-                for j in range(layer_in.classes):
-                    class_index = _entry_index(0, lw, lh, layer_in.outputs, layer_in.classes,
-                                               layer_in.coords, n*lw*lh + i, layer_in.coords+1+j)
-                    prob = scale*tvm_out[class_index]
-                    probs[index][j] = prob if prob > thresh else 0
-                    max_element = max(max_element, prob)
-                probs[index][layer_in.classes] = max_element
-
-    _correct_region_boxes(boxes, lw*lh*layer_in.n, imw, imh, netw, neth, relative)
-    return boxes, probs
-
-
-def do_nms_sort(boxes, probs, total, classes, thresh):
-    "Does the sorting based on the threshold values"
-    SortableBbox = namedtuple('SortableBbox', ['index_var', 'class_var', 'probs'])
+def _get_box(data, biases, n, location, lw, lh, w, h):
+    bx = (location[2] + data[location[0]][0][location[1]][location[2]]) / lw
+    by = (location[1] + data[location[0]][1][location[1]][location[2]]) / lh
+    bw = np.exp(data[location[0]][2][location[1]][location[2]]) * biases[2*n] / w
+    bh = np.exp(data[location[0]][3][location[1]][location[2]]) * biases[2*n+1] / h
+    return Box(bx, by, bw, bh)
 
-    s = [SortableBbox(0, 0, []) for i in range(total)]
-    for i in range(total):
-        s[i] = s[i]._replace(index_var=i)
-        s[i] = s[i]._replace(class_var=0)
-        s[i] = s[i]._replace(probs=probs)
+def _get_yolo_detections(l, im_w, im_h, netw, neth, thresh, relative, dets):
+    data = l['output']
+    np.set_printoptions(threshold=np.nan)
```

Comment:
np.set_printoptions(threshold=np.nan)-> this is needed?

Reply:
Yes it is. I will remove it.